test/batch-submitter/transaction-batch-submitter.spec.ts

```diff
@@ -282,3 +297,66 @@ describe('TransactionBatchSubmitter', () => {
     })
   })
 })
+
+describe('TransactionBatchSubmitter to Ganache', () => {
+  let signer
+  const server = ganache.server({
+    default_balance_ether: 420,
+    blockTime: 2_000
+  })
+  const provider = new Web3Provider(ganache.provider())
+
+  before(async () => {
+    await server.listen(3001)
+    signer = await provider.getSigner()
+  })
+
+  after(async () => {
+    await server.close()
+  })
+
+  // only write test for getReceiptWithResubmission
+  it('should resubmit a transaction if it is not confirmed', async () => {
+    let firstGasPrice
+    const numConfirmations = 2
+    const sendTxFunc = async (gasPrice) => {
+      // only set firstGasPrice when it's undefined
+      firstGasPrice = firstGasPrice === undefined ? gasPrice : firstGasPrice
+
+      // console.debug(`retried gasPrice: ${gasPrice}`)
+      const tx = signer.sendTransaction({
+        to: DECOMPRESSION_ADDRESS,
+        value: 88,
+        nonce: 0,
+        gasPrice
+      })
+
+      // try mining a block every time a transaction is submitted
+      await provider.send('evm_mine', [])
```

Comment:
I dont think we want to mine a block when a transaction is submitted so that there is an opportunity for the transaction to be replaced. I think what we want is to after the call to `getReceiptWithResubmission`, do an await `sleep` for 1.5 seconds (greater than resubmission timeout).

```js
function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms))
}
```

We can add a counter at the top of this `it` block that gets incremented as well as an array that the gas price gets pushed to from within the `sendTxFunc` and then after the sleep we can look at those values and make some assertions on them, ie that the counter was incremented (meaning `sendTxFunc` was called) as well as looking at the gas prices array to make sure they are increasing

Reply:
updated this, and just used the length of the gasPrices array as the counter :)